pkg/vgmanager/vgmanager_controller.go

```diff
@@ -243,3 +284,32 @@ func ToleratesTaints(tolerations []corev1.Toleration, taints []corev1.Taint) boo
 	}
 	return true
 }
+
+func (r *VGReconciler) updateStatus(ctx context.Context, status string, reason string, instance *lvmv1alpha1.LVMVolumeGroup) error {
+
+	found := false
+
+	nodeStatuses := instance.Status.NodeStatus
+	for _, nodeStatus := range nodeStatuses {
+		if nodeStatus.Node == r.Nodename {
+			found = true
+			nodeStatus.Status = status
+			nodeStatus.Reason = reason
+		}
```

Comment:
`break` for loop once `NodeName` is found.

Reply:
Done